### ISSUE TITLE
If a with statement’s target is a global variable that was assigned before, a false “redefinition of unused variable” is reported.

### ISSUE THREAD

**Issue by manueljacob** [astral-sh/ruff#14257]
```python
from contextlib import nullcontext

def f():
    global glob
    glob = 0
    assert g() == 0
    with nullcontext(enter_result=1) as glob:
        assert g() == 1

def g():
    return glob
```

With ruff 0.7.3:

```
test.py:8:41: F811 Redefinition of unused `glob` from line 6
  |
6 |     glob = 0
7 |     assert g() == 0
8 |     with nullcontext(enter_result=1) as glob:
  |                                         ^^^^ F811
9 |         assert g() == 1
  |
  = help: Remove definition: `glob`

Found 1 error.
```

However, `glob` is used by `g()`, so neither of the assignments can be removed.

I’m aware that according to [the Python documentation](https://docs.python.org/3/reference/simple_stmts.html#the-global-statement), “names listed in a [global](https://docs.python.org/3/reference/simple_stmts.html#global) statement must not be defined as formal parameters, or as targets in [with](https://docs.python.org/3/reference/compound_stmts.html#with) statements or [except](https://docs.python.org/3/reference/compound_stmts.html#except) clauses, or in a [for](https://docs.python.org/3/reference/compound_stmts.html#for) target list, [class](https://docs.python.org/3/reference/compound_stmts.html#class) definition, function definition, [import](https://docs.python.org/3/reference/simple_stmts.html#import) statement, or variable annotation”. However, that is currently not enforced by CPython or PyPy. There could be a ruff rule that reports an error in this case, but that’s a separate issue.

**Comment by MichaReiser** [comment]
@AlexWaygood would you mind taking a look at this issue?
